Upgrade orc version

galipremsagar opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

nehaljwani commented:
@galipremsagar You'll need to backport https://github.com/noirello/pyorc/commit/af5a7cd6036520b8c4cebd259f5ca5fc2fcdbe7c as a patch to this recipe to make the build pass. And I think you should also unpin the version from the recipe and let the global pinning take care of it. That way, whenever there is an update to orc, the bot will automatically open a PR here.

galipremsagar commented:
> @galipremsagar You'll need to backport [noirello/pyorc@af5a7cd](https://github.com/noirello/pyorc/commit/af5a7cd6036520b8c4cebd259f5ca5fc2fcdbe7c) as a patch to this recipe to make the build pass. And I think you should also unpin the version from the recipe and let the global pinning take care of it. That way, whenever there is an update to orc, the bot will automatically open a PR here.

Thanks, @nehaljwani Addressed your reviews.

jakirkham commented on recipe/patches/0001-Fix-software-version-test.patch:
```diff
@@ -1,3 +1,5 @@
+https://github.com/conda-forge/pyorc-feedstock/pull/18#issuecomment-1012846145
+
```
Ah would suggest adding this to `meta.yaml` as a comment. Also would be worth including the link to the commit

galipremsagar replied:
Done 👍

galipremsagar commented on recipe/patches/0001-Fix-software-version-test.patch:
```diff
@@ -0,0 +1,35 @@
+https://github.com/conda-forge/pyorc-feedstock/pull/18#issuecomment-1012846145
```
```suggestion
https://github.com/noirello/pyorc/commit/af5a7cd6036520b8c4cebd259f5ca5fc2fcdbe7c
```

galipremsagar commented on recipe/meta.yaml:
```diff
@@ -5,6 +5,8 @@ package:
 source:
   url: https://pypi.io/packages/source/p/pyorc/pyorc-0.5.0.tar.gz
   sha256: 89c4c708c653d0fe90f251fbcc4a6057daf7cf63c23dfe3d644d9ed2603e296d
+  patches:
+    - patches/0001-Fix-software-version-test.patch
```
```suggestion
  patches:
    - patches/0001-Fix-software-version-test.patch   # https://github.com/conda-forge/pyorc-feedstock/pull/18#issuecomment-1012846145
```

jakirkham commented on recipe/meta.yaml:
```diff
@@ -5,6 +5,8 @@ package:
   url: https://pypi.io/packages/source/p/pyorc/pyorc-0.5.0.tar.gz
   sha256: 89c4c708c653d0fe90f251fbcc4a6057daf7cf63c23dfe3d644d9ed2603e296d
+  patches:
+    - patches/0001-Fix-software-version-test.patch   # https://github.com/conda-forge/pyorc-feedstock/pull/18#issuecomment-1012846145
```
```suggestion
    # https://github.com/conda-forge/pyorc-feedstock/pull/18#issuecomment-1012846145
    - patches/0001-Fix-software-version-test.patch
```

jakirkham commented:
@conda-forge/pyorc, this is ready for review! 😄

noirello reviewed (approved):
Thank you, looks great. 

I put the pin because of that broken test. Thought I'll remove it with 0.6.0. But this works just as well. 😄

jakirkham commented:
Thanks @noirello! 😄

Yeah that makes sense.

We were just discussing over in issue ( https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/2400 ) that we can now leverage some of conda-forge's automation to get new PRs for `orc` updates added automatically with this change. So hopefully that makes things a bit smoother for future `orc` releases 🙂

jakirkham commented on recipe/meta.yaml:
```diff
@@ -5,6 +5,9 @@ package:
+    # https://github.com/conda-forge/pyorc-feedstock/pull/18#issuecomment-1012846145
+    - patches/0001-Fix-software-version-test.patch
 
 build:
   number: 1
```
We just realized we forgot to bump the build number 🤦‍♂️

PR ( https://github.com/conda-forge/pyorc-feedstock/pull/19 ) should address this
